internal/metadataproviders/openshift/metadata.go

```diff
@@ -0,0 +1,180 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openshift // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/metadataproviders/openshift"
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+// Provider gets cluster metadata from Openshift.
+type Provider interface {
+	K8SClusterVersion(context.Context) (string, error)
+	OpenShiftClusterVersion(context.Context) (string, error)
+	Infrastructure(context.Context) (*InfrastructureMetadata, error)
+}
+
+// NewProvider creates a new metadata provider.
+func NewProvider(address, token string) Provider {
+	return &openshiftProvider{
+		address: address,
+		token:   token,
+		client:  &http.Client{},
+	}
+}
+
+type openshiftProvider struct {
+	client  *http.Client
+	address string
+	token   string
+}
+
+func (o *openshiftProvider) mkOCPReq(ctx context.Context, endpoint, target string) (*http.Request, error) {
```

Comment:
what is this function doing? the name looks weird to me

Reply:
It generates an http get request including an Authorization header to the given resource status.

Reply:
It will be more readable, anyway is an small function I read it and figured it out, but yeah better name will help